packages/strapi-plugin-users-permissions/admin/src/containers/AuthPage/index.js

```diff
@@ -40,73 +41,90 @@ import styles from './styles.scss';
 
 export class AuthPage extends React.Component { // eslint-disable-line react/prefer-stateless-function
   componentDidMount() {
-    const params = this.props.location.search ? replace(this.props.location.search, '?code=', '') : this.props.match.params.id;
-    this.props.setForm(this.props.match.params.authType, params);
+    const { location: { search }, match: {params: { authType, id } }, setForm } = this.props; 
+    const params = search ? replace(search, '?code=', '') : id;
+    auth.clearAppStorage();
+    setForm(authType, params);
   }
 
-  componentWillReceiveProps(nextProps) {
-    if (this.props.match.params.authType !== nextProps.match.params.authType) {
-      const params = nextProps.location.search ? replace(nextProps.location.search, '?code=', '') : nextProps.match.params.id;
-      this.props.setForm(nextProps.match.params.authType, params);
-      this.props.hideLoginErrorsInput(false);
+  componentDidUpdate(prevProps) {
+    const { hideLoginErrorsInput, location: { search }, match: { params :{ authType, id }}, setForm, submitSuccess } = this.props;
```

Comment:
This is hard to read. Can we use multi-line to make more visible what is happening here?. Plus, avoiding long lines...

```suggestion
    const {
      hideLoginErrorsInput,
      location: { search },
      match: {
        params: { authType, id }
      },
      setForm,
      submitSuccess
    } = this.props; 
```

Or maybe do it in two steps to destruturing nested properties 🤷‍♂️

Reply:
For your information, i will add prettier in next few days to prevent this kind of feedback from code review.